packages/nuka/src/slider-list.ts

```diff
@@ -93,45 +95,85 @@ const getPositioning = (
 
   // Special-case this. It's better to return undefined rather than a
   // transform of 0 pixels since transforms can cause flickering in chrome.
-  if (move === 0 && horizontalMove === 0) {
+  if (draggedOffset === 0 && slideBasedOffset === 0) {
     return undefined;
   }
 
-  const draggableMove = move
-    ? `calc(${horizontalMove}% - ${move}px)`
-    : `${horizontalMove}%`;
-  return `translate3d(${draggableMove}, 0, 0)`;
+  let clampOffsets: number[] | null = null;
+  if (clampIndices) {
+    clampOffsets = clampIndices.map((index) =>
+      getPercentOffsetForSlide(
+        slideCount,
+        initialValue,
+        index,
+        cellAlign,
+        wrapAround
+      )
+    );
+  }
+
+  const clampedDraggedOffset = clampOffsets
+    ? // Offsets are seemingly backwards because the rightmost slide creates
+      // the most negative translate value
+      `clamp(${clampOffsets[1]}%, ${draggedOffset}px, ${clampOffsets[0]}%)`
+    : `${draggedOffset}px`;
+
+  return `translate3d(${
+    draggedOffset ? clampedDraggedOffset : `${slideBasedOffset}%`
+  }, 0, 0)`;
 };
 
 export const getSliderListStyles = (
   children: ReactNode | ReactNode[],
   currentSlide: number,
   isAnimating: boolean,
-  slidesToShow?: number,
-  cellAlign?: 'left' | 'right' | 'center',
-  wrapAround?: boolean,
-  speed?: number,
-  move?: number,
+  slidesToShow: number,
+  cellAlign: Alignment,
+  wrapAround: boolean,
+  speed: number,
+  draggedOffset: number,
+  slidesToScroll: number,
+  scrollMode: ScrollMode,
+  disableEdgeSwiping: boolean,
   slideAnimation?: 'fade' | 'zoom'
 ): CSSProperties => {
-  const count = React.Children.count(children);
+  const slideCount = React.Children.count(children);
+
+  const width = getSliderListWidth(slideCount, slidesToShow, wrapAround);
+
+  // When disableEdgeSwiping=true, we recycle dot index generation to determine
+  // the leftmost and rightmost indices used, to be used in calculating the
+  // x-translation values we need to limit to.
+  let clampIndices: number[] | null = null;
+  if (disableEdgeSwiping && !wrapAround) {
+    const dotIndexes = getDotIndexes(
+      slideCount,
+      slidesToScroll,
+      scrollMode,
+      slidesToShow,
+      wrapAround,
+      cellAlign
+    );
+    clampIndices = [dotIndexes[0], dotIndexes[dotIndexes.length - 1]];
+  }
 
-  const width = getSliderListWidth(count, slidesToShow, wrapAround);
   const positioning = getPositioning(
-    cellAlign || Alignment.Left,
-    slidesToShow || 1,
-    count,
+    cellAlign,
+    slidesToShow,
+    slideCount,
     currentSlide,
     wrapAround,
-    move
+    draggedOffset,
+    clampIndices
   );
 
   return {
     width,
     textAlign: 'left',
+    userSelect: 'auto',
     transition:
-      isAnimating && slideAnimation !== 'fade'
-        ? `${speed || 500}ms ease 0s`
+      draggedOffset === 0 && isAnimating && slideAnimation !== 'fade'
+        ? `${speed}ms cubic-bezier(0.16, 1, 0.3, 1) 0s`
```

Comment:
This is one of the iffier changes in the PR - changing the easing function used will make the slide changes look snappier / less relaxed than before. The goal of changing this was to make transitions starting from the end of a swipe look less stilted - `ease` has a slow buildup to its max speed which, coming off of a swipe, is unnatural in that it slows down and speeds up again. This new easing function does not have the slow buildup at the front, avoiding that issue, but, again, makes it feel "snappy" where it may not have before.